controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,138 @@
+    Supervisor* supervisor = new Supervisor();
+    Node* target           = supervisor->getFromDef("RED_1");
+    std::cout << "b" << std::endl;
+    // Get the time step of the current world.
+    int timeStep = (int) supervisor->getBasicTimeStep();
+
+    std::cout << "c" << std::endl;
+
+    // Add the motion manager to make the robot stand up and look around
+    managers::RobotisOp2MotionManager* mMotionManager = new managers::RobotisOp2MotionManager(supervisor);
+    mMotionManager->playPage(54);
+
+    // Generate random seed
+    srand(time(NULL));
+
+    // Main loop:
+    while (supervisor->step(timeStep) != -1) {
+
+        // Grab the current translation field of the robot to modify
+        Field* target_translation_field = target->getField("translation");
+        // Convert the field to a vector to output to console
+        const double* target_translation_vec = target_translation_field->getSFVec3f();
+
+        // Output current location
+        std::cout << "Location: " << std::endl;
+        std::cout << "X: " << target_translation_vec[0];
+        std::cout << " Y: " << target_translation_vec[1];
+        std::cout << " Z: " << target_translation_vec[2] << std::endl;
+
+        // Prepare new location
+
+        /* 0, 0, 0 is centre of the playing field.
+         *     X ranges from -5.4 - 5.4, the positive value on the left hand side when looking
+         *     from red
+         *     Y ranges from -8 - 8, the positive value on the red goal side
+         *     Z should be set to 3.2 to be on the ground
+         *     If the robot teleports into an existing object it may run into issues for that
+         *     image only, after resetPhysics is should return to a regular state
+         */
+
+        double newPos[3];
+        newPos[0] = 5.4 - (rand() % 1080) / 100;  // target_translation_vec[0] + 1;
+        newPos[1] = 8 - (rand() % 1600) / 100;    // target_translation_vec[1] + 0;
+        newPos[2] = 0.32;                         // target_translation_vec[2] + 0;
+
+        // Set new location
+        target_translation_field->setSFVec3f(newPos);
+
+        // Grab the current rotation field of the robot to modify
+        Field* target_rotation_field = target->getField("rotation");
+        // Convert the field to a vector to output to console
+        const double* target_rotation_vec = target_rotation_field->getSFRotation();
+
+        // Loop 4 times, one for each direction to face
+        for (int i = 0; i < rotationsSize; i++) {
+            // Output current rotation
+            std::cout << "Rotation: " << std::endl;
+            std::cout << "X: " << target_rotation_vec[0];
+            std::cout << " Y: " << target_rotation_vec[1];
+            std::cout << " Z: " << target_rotation_vec[2];
+            std::cout << " α: " << target_rotation_vec[3] << std::endl;
+
+            /* Prepare new rotation. These are saved in rotations matrix as the axis-angle
```

Comment:
_Usually_ we use // comments instead of /* comments, with the exception of javadoc style /** comments.